site/topics/programming/assembly.rst

```diff
@@ -22,6 +22,53 @@ Assembly Language
 Assembler
 =========
 
+* An assembler is a tool to convert assembly language to machine code
+* An assembly language is a very low level programming language
+
+    * Often referred to as *assembly*
+    * Assembly is strongly tied to a specific system design, the underlying hardware, and its machine language
```

Comment:
Could maybe link to [ARM's instruction set](https://azeria-labs.com/arm-instruction-set-part-3/) for a basic real world example.

Reply:
i love it